packages/runtime/test/src/runtime.ts

```diff
@@ -286,4 +293,164 @@ describe("Algorand Standard Assets", function () {
     assert.equal(john.getAssetHolding(assetId)?.amount, 0);
     assert.equal(alice.getAssetHolding(assetId)?.amount, initialAliceAssets + initialJohnAssets);
   });
+
+  it("should throw error if asset is not found while modifying", () => {
+    expectTealError(
+      () => runtime.modifyAsset(john.address, 120, modFields, {}),
+      ERRORS.ASA.ASSET_NOT_FOUND
+    );
+  });
+
+  it("should modify asset", () => {
+    const assetId = runtime.createAsset('gold',
+      { creator: { name: "john", addr: john.address, sk: john.account.sk } });
+
+    runtime.modifyAsset(elon.address, assetId, modFields, {});
+
+    const res = runtime.getAssetDef(assetId);
+    assert.equal(res.manager, bob.address);
+    assert.equal(res.reserve, bob.address);
+    assert.equal(res.clawback, john.address);
+    assert.equal(res.freeze, john.address);
+  });
+
+  it("Blank field test, should not modify asset because field is set to blank", () => {
+    const assetId = runtime.createAsset('silver',
+      { creator: { name: "john", addr: john.address, sk: john.account.sk } });
+
+    const modFields: AssetModFields = {
+      manager: bob.address,
+      reserve: bob.address,
+      clawback: john.address,
+      freeze: alice.address
+    };
+
+    expect(() => {
+      runtime.modifyAsset(elon.address, assetId, modFields, {});
+    }).to.throw("Cannot reset a blank address");
```

Comment:
Can we use `assert.throws` ? Since in the other tests we are using `assert`

Reply:
this is more intuitive and clear

Reply:
Ok. Then can we update other tests (in ASA) as well? Currently there is mix up of `asset.throws` and `expect.to.throw` here

Reply:
changed these errors to ASA errors, using expectTealError is better

Reply:
Yes, but it should be `expectRuntimeError` (TEAL is not associated here). We should update this in restructuring

Reply:
Yes that's why added that task. Not possible to fix all things in one big task

Reply:
I know, was confirming so that we don't miss it.